proposals/drafts/elections.rst

```diff
@@ -43,46 +255,502 @@ extras
+
+is_runoff
+    Indicates a contest to decide a prior contest that ended with no candidate receiving a majority of the votes (boolean).
+
+is_unexpired_term
+    Indicates that the former public office holder vacated the post before serving a full term (boolean).
+
+number_elected
+    **optional**
+    Number of candidates that are elected in the contest, i.e. 'N' of N-of-M (integer).
+
+post_ids
+    References to the ``Posts`` representing the public offices for which the candidates are competing. If multiple, the primary post should be listed first, e.g., the id for the President post should be listed before the id for Vice-President.
+
+party_id
+    **optional**
+    If the contest is among candidates of the same political party, e.g., a partisan primary election, reference to that ``Party``.
+
+votes_allowed
+    **optional**
+    Maximum number of votes/write-ins per voter in this contest.
+
+
+PartyContest
+------------
+
+A subclass of ``ContestBase`` which describes a contest in which the possible ballot selections are all political parties. These could include contests in which straight-party selections are allowed, or party-list contests (although these are more common outside of the United States).
+
+id
+    Open Civic Data-style id in the format ``ocd-partycontest/{{uuid}}``.
+
+
+RetentionContest
+----------------
+
+A subclass of ``BallotMeasureContest`` that represents a contest where a candidate is retained in a position, e.g. a judicial retention or recall election.
+
+id
+    Open Civic Data-style id in the format ``ocd-retentioncontest/{{uuid}}``.
+
+candidacy_id
+    Reference to the ``Candidacy`` of the person who will either retain or lose a ``Post`` as a result of the contest.
 
-posts
-    References to the ``Post`` that this contest is for
+post_id
+    Reference to the ``Post`` which the candidate will either retain or lose as a result of the contest.
+
+
+Candidacy
+---------
+
+Represents a person who is a candidate in a particular ``CandidateContest``. If a candidate is running in multiple contests, each contest must have its own ``Candidate`` object. ``Candidate`` objects may not be reused between contests.
+
+id
+    Open Civic Data-style id in the format ``ocd-candidacy/{{uuid}}``.
+
+ballot_name
+    The candidate's name as it will be displayed on the official ballot, e.g. "Ken T. Cuccinelli II" (string).
+
+committee_id
+    **optional**
+    Reference to the ``Committee`` (see OCDEP: Campaign Finance Filings) that represents the candidate's campaign committee for the contest.
+
+filed_date
+    **optional**
+    Specifics when the candidate filed for the contest (date).
+
+is_incumbent
+    **optional**
+    Indicates whether the candidate is the incumbent for the office associated with the contest.
+
+is_top_ticket
+    **optional**
+    Indicates whether the candidate is the top of a ticket that includes multiple candidates (boolean).
```

Comment:
Added some examples to this definition in the [other PR](https://github.com/opencivicdata/docs.opencivicdata.org/pull/64).